src/k8s/pkg/k8sd/features/cilium/gateway.go

```diff
@@ -15,12 +15,21 @@ import (
 // ApplyGateway will rollout restart the Cilium pods in case any Cilium configuration was changed.
 // ApplyGateway returns an error if anything fails.
 func ApplyGateway(ctx context.Context, snap snap.Snap, gateway types.Gateway, network types.Network, _ types.Annotations) error {
+	if !network.GetEnabled() && gateway.GetEnabled() {
+		return fmt.Errorf("cilium gateway requires network to be enabled")
+	}
```

Comment:
Interesting. I don't think I considered this in my PR 🤔

Reply:
This is irrelevant for the context of this PR, and is also currently handled elsewhere https://github.com/canonical/k8s-snap/blob/5f34bb08c02c59d8df6287aef8a151ab96f45f86/src/k8s/pkg/k8sd/types/cluster_config_validate.go#L33-L45

I would recommend not doing the same check in multiple places around the code, because that would be incredibly hard to then track down and adjust the code in the future.

For now, please revert this

Reply:
Yes, I wanted to move this to the feature implementation because it's not applicable for contour...

Reply:
Ok, I will remove it from this PR 👍🏼